angular/projects/researchdatabox/form/src/app/component/textarea.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input } from '@angular/core';
+import { FormFieldBaseComponent, FormFieldCompMapEntry, FormFieldModel } from "@researchdatabox/portal-ng-common";
+import { TextareaComponentConfig } from '@researchdatabox/sails-ng-common/dist/src/config/component/textarea.model';
+import { get as _get, isUndefined as _isUndefined, isEmpty as _isEmpty } from 'lodash-es';
+
+export class TextareaModel extends FormFieldModel<string> {
+}
+
+@Component({
+  selector: 'redbox-textarea',
+  template: `
+    @if (isVisible) {
+      <ng-container *ngTemplateOutlet="getTemplateRef('before')" />
+      <textarea [formControl]="formControl"
+        class="form-control"
+        [class.is-valid]="isValid"
+        [class.is-invalid]="!isValid"
+        [attr.required]="isRequired === true ? true : null"
+        [attr.disabled]="isDisabled ? 'true' : null"
+        [attr.readonly]="isReadonly ? 'true' : null"
+        [attr.title]="tooltip ? tooltip : tooltipPlaceholder"
+        [attr.rows]="rows"
+        [attr.cols]="cols"
+        [attr.placeholder]="placeholder"></textarea>
+      <ng-container *ngTemplateOutlet="getTemplateRef('after')" />
+    }
+  `,
+  standalone: false
+})
+export class TextareaComponent extends FormFieldBaseComponent<string> {
+  protected override logName: string = "TextareaComponent";
+  public tooltip:string = '';
+  public tooltipPlaceholder:string = 'placeholder';
+  public rows:number = 0;
+  public cols:number = 0;
```

Comment:
We discussed not using 0 as the defaults for these as they're not valid values as far as the browser is concerned. I think we agreed on 20 and 2 for the values but I think undefined as the default value is fine as well as that will remove the attribute unless a value is set.